### PR TITLE
feat: target configuration for action keymaps

### LINKED ISSUE #428: Remap key to key combination
I'm thinking how to bind y to the equivalent of 'a 5'. So, current song, copy link.
I think either having it so you can bind to a sequence of commands, or bind a key to a sequence of keys would be nice.
Something like
[[keymaps]]
command = "ShowActionsOnCurrentTrack; CopyTrackLink"
key_sequence = "y"
or
[[keymaps]]
command = "a 5" // maybe a different name instead of command? idk
key_sequence = "y"

### PR DESCRIPTION
**Pull request by juliamertz**
Actually fixes #428 this time 🙃

### CODE REVIEW

File: spotify_player/src/config/keymap.rs [hunk 1]
```diff
@@ -25,6 +25,7 @@ pub struct Keymap {
 /// A keymap that triggers an `Action` when a key sequence is pressed
 pub struct ActionMap {
     pub key_sequence: KeySequence,
+    pub target: Option<ActionTarget>,
```
Comment (aome510) [inline spotify_player/src/config/keymap.rs:28]:
```suggestion
     #[serde(default)]
    pub target: ActionTarget,
```

File: spotify_player/src/event/page.rs [hunk 2]
```diff
@@ -194,7 +194,7 @@ fn handle_key_sequence_for_search_page(
                 CommandOrAction::Command(command) => {
                     window::handle_command_for_artist_list_window(command, artists, &data, ui)
                 }
-                CommandOrAction::Action(action) => {
+                CommandOrAction::Action(action, _) => {
```
Comment (aome510) [inline spotify_player/src/event/page.rs:197]: should this be

```suggestion
                CommandOrAction::Action(action, ActionTarget::SelectedItem) => {
```
?

Same with other matches in this file

File: spotify_player/src/event/mod.rs [hunk 3]
```diff
@@ -105,8 +108,13 @@ fn handle_key_event(
 
     // if the key sequence is not handled, let the global command handler handle it
```
Comment (aome510) [inline spotify_player/src/event/mod.rs:109]:
```suggestion
    // if the key sequence is not handled, let the global handler handle it
```

File: spotify_player/src/event/mod.rs [hunk 4]
```diff
@@ -346,6 +383,40 @@ fn handle_show_actions_on_artist(
...
+        let data = state.data.read();
+
+        let currently_playing = match player.current_playing_track() {
+            Some(track) => track,
+            None => anyhow::bail!("No currently playing track"),
+        };
+        let track = match Track::try_from_full_track(currently_playing.clone()) {
+            Some(track) => track,
+            None => anyhow::bail!("No currently playing track"),
+        };
```
Comment (aome510) [inline spotify_player/src/event/mod.rs:406]: normally, we don't return error when there is nothing to act on when handling a key event, e.g. no playing track. See how `Command::ShowActionsOnCurrentTrack` is handled in `event/mod.rs`:

```rust
  if let Some(track) = state.player.read().current_playing_track() {
      if let Some(track) = Track::try_from_full_track(track.clone()) {
          let data = state.data.read();
          ...
```

**Review by aome510 — COMMENTED** [review]
Thanks for the PR. I left a few comments other looks good.

File: spotify_player/src/event/page.rs (hunk 2 above)
Comment (juliamertz) [reply spotify_player/src/event/page.rs:197]: You're right, this was an oversight on my part

**Comment by juliamertz** [comment]
Should all be fixed now, only formatting caused a conflict

**Review by aome510 — APPROVED** [review]

**Pull request merged**